src/main/kotlin/uk/gov/justice/digital/hmpps/prisonercontactregistry/controller/PrisonerContactController.kt

```diff
@@ -69,11 +72,82 @@ class PrisonerContactController(
     @Parameter(description = "Query by Person Identifier (NOMIS Person ID)", example = "9147510")
     personId: Long?,
     @RequestParam(value = "withAddress", required = false)
-    @Parameter(description = "by default returns addresses for all contacts, set to false if contact addresses not needed.", example = "false")
+    @Parameter(
+      description = "by default returns addresses for all contacts, set to false if contact addresses not needed.",
+      example = "false",
+    )
     withAddress: Boolean? = true,
   ): List<ContactDto> {
     log.debug("Prisoner: $prisonerId, Type: $contactType, Person: $personId, withAddress = $withAddress")
-    return orderByLastNameAndThenFirstName(contactService.getContactList(prisonerId, contactType, personId, withAddress))
+    return orderByLastNameAndThenFirstName(
+      contactService.getContactList(
+        prisonerId,
+        contactType,
+        personId,
+        withAddress,
+      ),
+    )
+  }
+
+  @PreAuthorize("hasRole('PRISONER_CONTACT_REGISTRY')")
+  @GetMapping("{prisonerId}/approved/social/contacts/restrictions/banned/dateRange")
+  @Operation(
+    summary = "Get date range for visitors with ban restriction can visit a prisoner",
+    description = "Returns a date range for visitors with ban restriction can visit a prisoner",
+    responses = [
+      ApiResponse(
+        responseCode = "200",
+        description = "Date range returned",
+      ),
+      ApiResponse(
+        responseCode = "400",
+        description = "Incorrect request to Get date range for prisoner visitors",
+        content = [Content(mediaType = "application/json", schema = Schema(implementation = ErrorResponse::class))],
+      ),
+      ApiResponse(
+        responseCode = "401",
+        description = "Unauthorized to access this endpoint",
+        content = [Content(mediaType = "application/json", schema = Schema(implementation = ErrorResponse::class))],
+      ),
+      ApiResponse(
+        responseCode = "403",
+        description = "Incorrect permissions to Get date range for prisoner visitors",
+        content = [Content(mediaType = "application/json", schema = Schema(implementation = ErrorResponse::class))],
+      ),
+      ApiResponse(
+        responseCode = "404",
+        description = "Prisoner, Visitor or Date range not found",
+        content = [Content(mediaType = "application/json", schema = Schema(implementation = ErrorResponse::class))],
+      ),
+    ],
+  )
+  fun getDateRangeForPrisonerVisitorsWithBanRestrictions(
+    @Schema(description = "Prisoner Identifier (NOMIS Offender No)", example = "A1234AA", required = true)
+    @PathVariable
+    prisonerId: String,
+    @RequestParam(value = "visitors", required = true)
+    @Parameter(description = "Visitor Ids of prisoner contacts", example = "9147510, 8431201")
+    visitorIds: List<Long>,
```

Comment:
should it take visitorIds or a single visitor as dateRange can differ between prisoners

Reply:
It takes a list of visitors, i believe as they all visit together on a single booking, we can't only evaluate 1 visitor, we must evaluate them as a group? @AledWEvans might be able to confirm this

Reply:
yes that's right. I forgot what the new endpoint is for.